discovery-provider/src/queries/download_csv.py

```diff
@@ -0,0 +1,205 @@
+from src.models.tracks.track import Track
+from src.models.tracks.track_route import TrackRoute
+from src.models.users.usdc_purchase import PurchaseType, USDCPurchase
+from src.models.users.usdc_transactions_history import (
+    USDCTransactionMethod,
+    USDCTransactionsHistory,
+    USDCTransactionType,
+)
+from src.models.users.user import User
+from src.models.users.user_bank import USDCUserBankAccount
+from src.utils.config import shared_config
+from src.utils.csv_writer import write_csv
+from src.utils.db_session import get_db_read_replica
+
+env = shared_config["discprov"]["env"]
+
+
+class DownloadPurchasesArgs(TypedDict):
+    buyer_user_id: int
+
+
+class DownloadSalesArgs(TypedDict):
+    seller_user_id: int
+
+
+class DownloadWithdrawalsArgs(TypedDict):
+    user_id: int
+
+
+# Get all purchases or sales for a given artist.
+def get_purchases_or_sales(user_id: int, is_purchases: bool):
+    db = get_db_read_replica()
+    with db.scoped_session() as session:
+        if is_purchases:
+            # Get all purchases for the given user and include the seller name and handle
+            query = (
+                session.query(
+                    USDCPurchase.content_type.label("content_type"),
+                    USDCPurchase.created_at.label("created_at"),
+                    USDCPurchase.amount.label("amount"),
+                    User.handle.label("seller_handle"),
+                    User.name.label("seller_name"),
+                )
+                .join(User, User.user_id == USDCPurchase.seller_user_id)
+                .filter(USDCPurchase.buyer_user_id == user_id)
+                .filter(User.is_current == True)
+            )
+        else:
+            # Get all sales for the given user and include the buyer name
+            query = (
+                session.query(
+                    USDCPurchase.content_type.label("content_type"),
+                    USDCPurchase.created_at.label("created_at"),
+                    USDCPurchase.amount.label("amount"),
+                    User.name.label("buyer_name"),
+                )
+                .join(User, User.user_id == USDCPurchase.buyer_user_id)
+                .filter(USDCPurchase.seller_user_id == user_id)
+                .filter(User.is_current == True)
+            )
+
+        # Include playlist titles
+        playlists_query = (
+            query
+            .filter(
+                or_(
+                    USDCPurchase.content_type == PurchaseType.playlist,
+                    USDCPurchase.content_type == PurchaseType.album,
+                )
+            )
+            .join(Playlist, Playlist.playlist_id == USDCPurchase.content_id)
+            .join(PlaylistRoute, PlaylistRoute.playlist_id == USDCPurchase.content_id)
+            .filter(Playlist.is_current == True)
+            .add_columns(
+                Playlist.playlist_name.label("content_title"),
+                PlaylistRoute.slug.label("slug"),
```

Comment:
I _think_ we should have relationships for these? Should be able to tell SQL Alchemy to eagerly load `Playlist.permalink` and avoid the custom join